test/check/check.go

```diff
@@ -468,11 +524,51 @@ func checkForDaemonSet(name string) error {
 }
 
 func checkForSecret(name string) error {
-	return framework.Global.Client.Get(context.Background(), types.NamespacedName{Name: name, Namespace: components.Namespace}, &corev1.Secret{})
+	secret := corev1.Secret{}
+	err := framework.Global.Client.Get(context.Background(), types.NamespacedName{Name: name, Namespace: components.Namespace}, &secret)
+	if err != nil {
+		return err
+	}
+
+	err = checkRelationshipLabels(secret.GetLabels(), "Secret", name)
+	if err != nil {
+		return err
+	}
+
+	return nil
 }
 
 func checkForMutatingWebhookConfiguration(name string) error {
-	return framework.Global.Client.Get(context.Background(), types.NamespacedName{Name: name}, &admissionregistrationv1.MutatingWebhookConfiguration{})
+	mutatingWebhookConfig := admissionregistrationv1.MutatingWebhookConfiguration{}
+	err := framework.Global.Client.Get(context.Background(), types.NamespacedName{Name: name}, &mutatingWebhookConfig)
+	if err != nil {
+		return err
+	}
+
+	err = checkRelationshipLabels(mutatingWebhookConfig.GetLabels(), "MutatingWebhookConfiguration", name)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func checkRelationshipLabels(labels map[string]string, kind, name string) error {
```

Comment:
can we unify `checkWorkloadRelationLabels` and `checkRelationshipLabels` by changing the input var `labels map[string]string` to be a list of `[]map[string]string`?
```suggestion
func checkRelationshipLabels(labelMapList []map[string]string, kind, name string) error {
```

You could then just iterate over the labels:
```
for labels := range labelMapList{
... // current checkRelationshipLabels() code
}
```

Reply:
I think it will be less readable,
i prefer the logic separation if possible, as one is the labels from the spec and one is the labels from the metadata
and this way its explicit instead of array of arrays

Reply:
I see what you mean, but imo the benefit would outcast the issues.
unifying it would make unit tests simpler, and would reduce redundant code.. 
at the end we are checking sets of labels of type `map[string]string`.

Reply:
as we spoke its not unit test, but e2e
anyhow, i did partially that, let me know if its good enough, (on next push),
i still kept the original `checkRelationshipLabels`, so it will be easier to call it when there is just one label to check, without creating an array